analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,179 @@
+            }
+
+            public override void Visit(SyntaxNode node)
+            {
+                if (!HasValidLoggerCall)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    base.Visit(node);
+                }
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                if (model.GetSymbolInfo(node, cancellationToken).Symbol is IMethodSymbol { ReceiverType: { } receiver } methodSymbol
+                    && receiver.DerivesOrImplements(KnownType.Microsoft_Extensions_Logging_ILogger))
+                {
+                    if (IsValidLogCall(node, methodSymbol))
+                    {
+                        HasValidLoggerCall = true;
+                    }
+                    else
+                    {
+                        invalidInvocations ??= new();
+                        invalidInvocations.Add(node.GetLocation());
+                    }
+                }
+                base.VisitInvocationExpression(node);
+            }
+
+            public override void VisitArgument(ArgumentSyntax node)
+            {
+                if (model.GetTypeInfo(node.Expression, cancellationToken).Type?.DerivesOrImplements(KnownType.Microsoft_Extensions_Logging_ILogger) is true)
+                {
+                    HasValidLoggerCall = true;
+                }
+                base.VisitArgument(node);
+            }
+
+            private bool IsValidLogCall(InvocationExpressionSyntax invocation, IMethodSymbol methodSymbol) =>
+                methodSymbol switch
+                {
+                    // Wellknown LoggerExtensions methods invocations
+                    { IsExtensionMethod: true } when methodSymbol.ContainingType.Is(KnownType.Microsoft_Extensions_Logging_LoggerExtensions) =>
+                        methodSymbol.Name switch
+                        {
+                            "LogInformation" or "LogWarning" or "LogError" or "LogCritical" => true,
+                            "Log" => IsPassingValidLogLevel(invocation, methodSymbol),
+                            "LogTrace" or "LogDebug" or "BeginScope" => false,
+                            _ => true, // Some unknown extension method on LoggerExtensions was called. Avoid FPs and assume it logs something.
+                        },
+                    { IsExtensionMethod: true } => true, // Any other extension method is assumed to log something to avoid FP.
+                    _ => // Instance invocations on an ILogger instance.
+                        methodSymbol.Name switch
+                        {
+                            "Log" => IsPassingValidLogLevel(invocation, methodSymbol),
+                            "IsEnabled" or "BeginScope" => false,
+                            _ => true, // Some unknown method on an ILogger was called. Avoid FPs and assume it logs something.
+                        },
```

Comment:
The indentation was broken. I fixed it in 923df2da3b06da2d68d47c4eeb75ee9c3517c7d0. Please have a second look, if this is okay.

Reply:
I'd say both were fine.